templates/backup-pruner.sh

```diff
@@ -0,0 +1,14 @@
+#jinja2: trim_blocks:False
+#!/bin/bash
+# This script runs all of the backup pruners in parallel, returning the output
+# of only the failing pruners.
+# Check {{ JOB_LOG }} for a summary of the most recent pruner run.
+set -e
+
+jobs='{% for item in TARSNAPPER_JOBS %}
+{{ item.name }}{% endfor %}'
+
+# Buffer all of the pruner output; if the pruner fails, forward along the
+# output to include in a cron email.
+wrapper='output=$(/usr/local/sbin/tarsnap-{}.sh expire 2>&1) || echo "$output"'
+echo "$jobs" | tail -n +2 | parallel --no-notice -j0 --joblog {{ JOB_LOG }} $wrapper
```

Comment:
@bdero The bit that I was missing is that GNU parallel (with the parameters you provided) won't have any output other than the output of the child processes.  Maybe you could add a comment pointing this out?  And you could also include that cron sends an email if and only if the cron job writes to stdout or stderr, regardless of the exit code.

Another reason this is hard to read is that it depends on the exact order of the different parsing and subsitution stages in shell command line parsing.  One way of improving this might be to add a flag to all the service-specific scripts to only print their output if they fail.  This would at least make this cron script more readable, but I'm not sure it's worth it overall, given that the whole thing should be refactored the next time we make any major changes.